### DISCUSSION TITLE
Why i always get the popup asking me to confirm overwriting .git/COMMIT_EDITMSG?

### DISCUSSION THREAD

**Discussion by shelper** [NeogitOrg/neogit#1473] [Q&A]
i use the default setting for neogit latest (commit 9bc4ee8) with minor changes in lazyvim as below:
```
	{
		"NeogitOrg/neogit",
		dependencies = {
			"nvim-lua/plenary.nvim", -- required
			"sindrets/diffview.nvim", -- optional - Diff integration
			"nvim-telescope/telescope.nvim", -- optional
		},
		opts = {
			integrations = {
				telescope = true,
				diffview = true,
				fzf_lua = nil,
			},
		},
		config = true,
	},
```

but whenever i press c c to commit i always get the pop up as shown in the figure below, even i manually delete the commit_editmsg right before i press c to commit, the file comes back and this popup shows up ...

Did i made any mistake? or is there a configuration for that .

![image](https://github.com/user-attachments/assets/b11f88a1-837f-427e-a69f-ad7303c2e597)

**Comment by CKolkey** [comment]
It's possible this is coming from another part of your config. One way to check would be to `:write` the buffer before closing it and see if the popup still appears.

**Reply by shelper** [reply]
you mean the buffer for the commit message? here is what i tried:
1. delete .git/COMMIT_EDITMSG
2. run `neogit commit`, no .git/COMMIT_EDITMSG at this moment
3. press 'c' in the splitted neogit window,  then this message popup, and .git/COMMIT_EDITMSG also got created
4. press Y to confirm the popup and edit the commit message, and then `:write", 
5. commit is there, also the newly created .git/COMMIT_EDITMSG
6. made some changes, go to step 2, same thing again

btw, when i am using lazygit in nvim (as configed by lazyvim) no such issue

**Comment by shelper** [comment]
well i think you are right, i tested just the neogit plugin and it does not have the popup...
really dont know where this pop up come from ...

**Comment by shelper** [comment]
turns out it's due to an autocmd i setup:

`vim.api.nvim_create_autocmd({ "FocusLost", "BufLeave" }, { pattern = { "*" }, command = "silent! wall", nested = true, })
`
once i delete this, no popup,
 but i just dont know how to fix it, Tried something like below but still not working

```
vim.api.nvim_create_autocmd({ "FocusLost", "BufLeave" }, {

pattern = { "*" },

callback = function()

local bufname = vim.fn.bufname()

local filetype = vim.bo.filetype

if filetype ~= "gitcommit" and not string.match(bufname, "/%.git/") then

vim.cmd("silent! wall")

end

end,

nested = true,

})

```

**Reply by CKolkey** [reply]
Oh, the filetype is `NeogitCommitMessage`, not `gitcommit`. You can find this in vim via `:echo &ft` at any time.

**Comment by parisni** [comment]
thanks for this issue, saves time to figure out what was going on. This workaround fixes it:
```lua
vim.api.nvim_create_autocmd("BufLeave", {
  pattern = "*", -- Trigger on all buffers
  callback = function()
    local filetype = vim.bo.filetype
    if not string.match(filetype, "^Neogit") then
      vim.cmd("silent! wall")
    end
  end,
  nested = true,
})

```
